packages/ckb-sdk-utils/src/address/index.ts

```diff
@@ -0,0 +1,75 @@
+import { bech32, blake160, hexToBytes } from '..'
+
+export enum AddressPrefix {
+  Mainnet = 'ckb',
+  Testnet = 'ckt',
+}
+
+export enum AddressType {
+  BinHash = '0x00',
+  BinIdx = '0x01',
+}
+
+export enum AddressBinIdx {
+  P2PH = 'P2PH',
+}
+
+export interface AddressOptions {
+  prefix: AddressPrefix
+  type: AddressType
+  binIdx: AddressBinIdx
+}
+
+export const defaultAddressOptions = {
+  prefix: AddressPrefix.Mainnet,
+  type: AddressType.BinIdx,
+  binIdx: AddressBinIdx.P2PH,
+}
+
+/**
+ * @description payload = type(01) | bin-idx("P2PH" => "50|32|50|40") | blake160-formatted pubkey
+ * @see https://github.com/nervosnetwork/ckb/wiki/Common-Address-Format
+ */
+export const toAddressPayload = (
+  blake160Pubkey: string | Uint8Array,
+  type: AddressType = AddressType.BinIdx,
+  params: AddressBinIdx = AddressBinIdx.P2PH
+): Uint8Array => {
+  if (typeof blake160Pubkey === 'string') {
+    return new Uint8Array([...hexToBytes(type), ...Buffer.from(params), ...hexToBytes(blake160Pubkey)])
+  }
+  return new Uint8Array([...hexToBytes(type), ...Buffer.from(params), ...blake160Pubkey])
+}
+
+export const blake160PubkeyToAddress = (
```

Comment:
I think this should not be called `blake160PublickeyToAddress`, as it's simply a `bech32` call.

The reason is `blake160` only applies to `P2PH` binIdx and `0x01` type. Now that we accept these two parameters it can take data other than `blake160` data.

Reply:
updated in https://github.com/nervosnetwork/ckb-sdk-js/pull/100/commits/32eaea566b87c8be9abd26419682bbfd1c746c64